### PR TITLE
Skip irrelevant submodules when building on Arch

### PR DESCRIPTION
**Pull request by Tea23**
## Description
The ffmpeg submodules are quite large, so pulling them in when building on Arch adds some unnecessary extra time to the build. When building on Arch, we're only interested in `ffpmeg-linux-x86_64` and the others are not used at all.

Arguably we could conditionally pull `ffpmeg-linux-aarch64` instead of its x86_64 counterpart when building for / on aarch64, but because `arch` doesn't specify aarch64 as a valid architecture for this package, there's no real need.

### Screenshot
N/A

### Issues Fixed or Closed
N/A

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [x] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [X] I want maintainers to keep my branch updated

### CODE REVIEW

**Comment by ReenigneArcher** [comment]
Thank you for your submission.

> Arguably we could conditionally pull ffpmeg-linux-aarch64 instead of its x86_64 counterpart when building for / on aarch64, but because arch doesn't specify aarch64 as a valid architecture for this package, there's no real need.

I was planning to enable this for aarch64, but as far as I could find all the dependencies are not available for arm64. Perhaps I am searching wrong, but when I search for `avahi` for example, it just shows x86_64... https://archlinux.org/packages/?sort=&q=avahi&maintainer=&flagged=

In fact, arm64/aarch64 is not even listed as a possible architecture.
image

**Comment by Tea23** [comment]
Strictly speaking, Arch only supports x86_64 and so they don't list alternative architectures on the official AUR. The `arch` parameter in a PKGBUILD specifies which architectures the package maintainer deems it to be compatible with, but the upstream distro itself will only support/endorse/etc x86_64. aarch64 and i686 are supplied by separate projects (https://archlinuxarm.org/ and https://archlinux32.org/).

Nothing stops you from defining i686 and/or aarch64 in `arch`, you just forfeit any distro level support and have to take it on yourself. If we *do* want this package to build on aarch64, we'd need to adjust the `prepare` section to conditionally pull the appropriate submodule.

**Comment by ReenigneArcher** [comment]
> Strictly speaking, Arch only supports x86_64 and so they don't list alternative architectures on the official AUR. The `arch` parameter in a PKGBUILD specifies which architectures the package maintainer deems it to be compatible with, but the upstream distro itself will only support/endorse/etc x86_64. aarch64 and i686 are supplied by separate projects (https://archlinuxarm.org/ and https://archlinux32.org/).
> 
> Nothing stops you from defining i686 and/or aarch64 in `arch`, you just forfeit any distro level support and have to take it on yourself. If we _do_ want this package to build on aarch64, we'd need to adjust the `prepare` section to conditionally pull the appropriate submodule.

Okay, would you mind making the adjustment, as well as adding `aarch64` to the `arch` parameter?

**Comment by Tea23** [comment]
@ReenigneArcher Sure, although you may want to remove i686 altogether, there only seems to be x86_64 and aarch64 ffpmeg submodules so sunshine may not build in i686 at all.

**Comment by ReenigneArcher** [comment]
> you may want to remove i686 altogether

That's fine with me as well.

File: packaging/linux/aur/PKGBUILD [hunk 1]
```diff
@@ -20,8 +20,24 @@ source=("$pkgname::git+@GITHUB_CLONE_URL@#commit=@GITHUB_COMMIT@")
...
+    elif [[ $CARCH == "aarch64" ]]; then
+        git -c submodule."ffmpeg-macos-x86_64".update=none \
+            -c submodule."ffmpeg-windows-x86_64".update=none \
+            -c submodule."ffmpeg-linux-x86_64".update=none \
+            -c submodule."ffmpeg-macos-aarch64".update=none \
+            submodule update --recursive --init
+
+# It's unlikely that someone could get this far on a system with an incorrect arch, but we should handle it anyway; pull everything
+    else
+        git submodule update --recursive --init
```
Comment (ReenigneArcher) [inline packaging/linux/aur/PKGBUILD:39]: Really no point in pulling the non linux ones... maybe set those to `update=none`?

File: packaging/linux/aur/PKGBUILD [hunk 2]
```diff
@@ -5,7 +5,7 @@ pkgname=@SUNSHINE_AUR_PKG@
 pkgver=@PROJECT_VERSION@@SUNSHINE_SUB_VERSION@
 pkgrel=1
 pkgdesc="@PROJECT_DESCRIPTION@"
-arch=('x86_64' 'i686')
+arch=('x86_64' 'i686' 'aarch64')
```
Comment (ReenigneArcher) [inline packaging/linux/aur/PKGBUILD:8]: Can you remove `i686` from here?

**Review by ReenigneArcher — CHANGES_REQUESTED** [review]

**Review by ReenigneArcher — APPROVED** [review]

**Comment by KuleRucket** [comment]
You have most commands in the if/else block repeating for every branch.

All of this could simply be outside the if block:
```
        git -c submodule."ffmpeg-macos-x86_64".update=none \
            -c submodule."ffmpeg-windows-x86_64".update=none \
            -c submodule."ffmpeg-macos-aarch64".update=none \
            submodule update --recursive --init
```

**Comment by ReenigneArcher** [comment]
@KuleRucket then it would grab non architecture specific submodules in all cases.

**Pull request merged**